# LiDAR cone detector configuration file
# Matt Young, 2022, UQRacing

lidar_processing:
  ########## TOPICS ##########
  # Topic to receive PointCloud2 LiDAR data from
  lidar_topic: "/lslidar_point_cloud"
  # Topic we receive CameraInfo messages from
  camera_info_topic: "/camera/camera_info"
  # Topic to publish uqr_msgs/DepthImage to
  lidar_depth_pub: "/lidar_processing/depth"

  ########## DEBUG TOPICS ##########
  # Topic to debug publish inpainting mask to. If empty, this is disabled.
  inpainting_debug_pub: "/lidar_processing/inpaintingDebug/compressed"
  # Topic we receive compressed camera frames from. Only for debug. If empty string, camera frames
  # will not be attempted to be acquired.
  camera_frame_topic: "/camera/image_rect_color/compressed"
  # Topic to publish depth images to. Compressed images only. Must end in /compressed.
  # This is **only** the image itself, and not the associated data to infer distances. Therefore
  # this is only useful in debugging.
  lidar_depth_image_pub: "/lidar_processing/depthImage/compressed"

  ########## PIPELINE SETTINGS ##########
  # Enable morphological operations. Basically increases pixel size (makes it more chunky). Fast.
  morphological: true
  # If true, publish a colour image (not useful for depth lookups, but good for debug/showing off).
  publishColour: false

  # Morphological operator kernel size (TODO implement this)
  morphKernelSize: 16

  ########## CAMERA PARAMETERS ##########
  # Extrinsic parameters
  # Rotation vector (x,y,z)
  rvec:
    - 0.0
    - -1.2
    - 0.0
  # Translation vector (x,y,z)
  tvec:
    - 0.0
    - 0.04
    - 0.03
